packages/flutter/lib/src/material/predictive_back_page_transitions_builder.dart

```diff
@@ -159,130 +265,363 @@ class _PredictiveBackGestureDetectorState extends State<_PredictiveBackGestureDe
+  @override
+  State<_PredictiveBackSharedElementPageTransition> createState() =>
+      _PredictiveBackSharedElementPageTransitionState();
+}
 
-    final bool isCurrent = getIsCurrent();
-    final Tween<double> xShiftTween =
-        isCurrent ? ConstantTween<double>(0) : Tween<double>(begin: xShift, end: 0);
-    final Animatable<double> scaleTween =
-        isCurrent
-            ? ConstantTween<double>(_scaleFullyOpened)
-            : TweenSequence<double>(<TweenSequenceItem<double>>[
-              TweenSequenceItem<double>(
-                tween: Tween<double>(begin: _scaleStartTransition, end: _scaleFullyOpened),
-                weight: _weightForStartState,
-              ),
-              TweenSequenceItem<double>(
-                tween: Tween<double>(begin: _scaleFullyOpened, end: _scaleFullyOpened),
-                weight: _weightForEndState,
-              ),
-            ]);
-    final Animatable<double> fadeTween =
-        isCurrent
-            ? ConstantTween<double>(_opacityFullyOpened)
-            : TweenSequence<double>(<TweenSequenceItem<double>>[
-              TweenSequenceItem<double>(
-                tween: Tween<double>(begin: _opacityFullyOpened, end: _opacityStartTransition),
-                weight: _weightForStartState,
+class _PredictiveBackSharedElementPageTransitionState
+    extends State<_PredictiveBackSharedElementPageTransition>
+    with SingleTickerProviderStateMixin {
+  // Constants as per the motion specs
+  // https://developer.android.com/design/ui/mobile/guides/patterns/predictive-back#motion-specs
+  static const double _kMinScale = 0.90;
+  static const double _kDivisionFactor = 20.0;
+  static const double _kMargin = 8.0;
+  static const double _kYPositionFactor = 0.1;
+
+  // Ideally this would match the curvature of the physical Android device being
+  // used, but that is not yet supported. Instead, this value is a best guess at
+  // a value that looks reasonable on most devices.
+  // See https://github.com/flutter/flutter/issues/97349.
+  static const double _kDeviceBorderRadius = 32.0;
+
+  // Eyeballed on a Pixel 9 running Android 16.
+  static const int _kCommitMilliseconds = 100;
+
+  // Since we don't know the device border radius, this provides a smooth
+  // transition between the default radius and the actual radius.
+  final Tween<double> _borderRadiusTween = Tween<double>(begin: _kDeviceBorderRadius, end: 0.0);
+
+  final Tween<double> _gapTween = Tween<double>(begin: _kMargin, end: 0.0);
+  final Tween<double> _scaleTween = Tween<double>(begin: _kMinScale, end: 1.0);
+  final Tween<double> _opacityTween = Tween<double>(begin: 1.0, end: 0.0);
+
+  late final AnimationController _commitController;
+  late final CurvedAnimation _commitAnimation;
+  late final Listenable _mergedAnimations;
+  late final Animation<double> _opacityAnimation;
+  late final Animation<double> _scaleAnimation;
+  late Animation<double> _scaleCommitAnimation;
```

Comment:
consider using a proxyAnimation to abstract away widget.animation and commit animation so that you don't have to recreate the animation every time phase changes